x-pack/plugins/fleet/common/types/models/agent_policy.ts

```diff
@@ -80,9 +86,19 @@ export interface FullAgentPolicyInput {
     [key: string]: unknown;
   };
   streams?: FullAgentPolicyInputStream[];
+  processors?: Array<FullAgentPolicyAddFields>;
```

Comment:
I'm wondering if we expect more types of processors in the future, maybe we can extend this later to have a generic processors interface and extend it with `AddFields`

Reply:
If I understand correctly looking at the processor docs, there is not standard structure for a processor. That's why I set the type of the processor as an array. I was thinking if we need to add more processors in the future, we can add those types in the array, for example `Array<FullAgentPolicyAddFields | AnotherType>`. I may be missing something though, so please let me know.